ssas/logger.go

```diff
@@ -22,48 +23,58 @@ type Event struct {
 }
 
 func init() {
-	Logger = logrus.New()
-	Logger.Formatter = &logrus.JSONFormatter{}
-	Logger.Formatter.(*logrus.JSONFormatter).TimestampFormat = time.RFC3339Nano
+
+	logInstance := logrus.New()
+	logInstance.SetFormatter(&logrus.JSONFormatter{TimestampFormat: time.RFC3339Nano}) //change to setformatter
 
 	filePath, success := os.LookupEnv("SSAS_LOG")
 	if success {
 		/* #nosec -- 0664 permissions required for Splunk ingestion */
 		file, err := os.OpenFile(filePath, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0664)
 
 		if err == nil {
-			Logger.SetOutput(file)
+			logInstance.SetOutput(file)
 		} else {
-			Logger.Info("Failed to open SSAS log file; using default stderr")
+			logInstance.Info("Failed to open SSAS log file; using default stderr")
 		}
 	} else {
-		Logger.Info("No SSAS log location provided; using default stderr")
+		logInstance.Info("No SSAS log location provided; using default stderr")
 	}
+
+	Logger = logInstance.WithFields(logrus.Fields{
+		"application": constants.Application,
+		"environment": os.Getenv("DEPLOYMENT TARGET"),
+		"version":     constants.Version})
+
+}
+
+type APILoggerEntry struct {
+	Logger logrus.FieldLogger
 }
 
-func mergeNonEmpty(data Event) *logrus.Entry {
-	var entry = logrus.NewEntry(Logger)
+func mergeNonEmpty(data Event) logrus.FieldLogger {
+	entry := &APILoggerEntry{Logger: Logger}
```

Comment:
Does this still need to create a new entry or is it okay to re-use the existing FieldLogger? This seems okay but I don't really know the difference 😅